systemtest/src/main/java/io/strimzi/systemtest/resources/ResourceManager.java

```diff
@@ -188,29 +198,32 @@ public static HelmClient helmClient() {
         assertNotNull(resource);
         assertNotNull(resource.getMetadata());
         assertNotNull(resource.getMetadata().getName());
+
+        // cluster role binding does not need namespace...
+        if (!(resource instanceof ClusterRoleBinding)) {
+            assertNotNull(resource.getMetadata().getNamespace());
+        }
+
         ResourceType<T> type = findResourceType(resource);
         assertNotNull(type);
 
-        while (!timeout.timeoutExpired()) {
-            T res = type.get(resource.getMetadata().getNamespace(), resource.getMetadata().getName());
-            if (condition.test(res)) {
-                LOGGER.info("Resource {} in namespace {} is {}!", resource.getMetadata().getName(), resource.getMetadata().getNamespace(), condition.toString());
-                return true;
-            }
-            try {
-                Thread.sleep(Duration.ofSeconds(1).toMillis());
-            } catch (InterruptedException e) {
-                LOGGER.info("Resource {} in namespace {} is not {}!", resource.getMetadata().getName(), resource.getMetadata().getNamespace(), condition.toString());
-                Thread.currentThread().interrupt();
-                return false;
-            }
-        }
-        T res = type.get(resource.getMetadata().getNamespace(), resource.getMetadata().getName());
-        boolean pass = condition.test(res);
-        if (!pass) {
-            LOGGER.info("Resource {} with with type {} failed condition check: {}", resource.getMetadata().getName(), type, resourceToString(res));
-        }
-        return pass;
+        boolean[] resourceReady = new boolean[1];
+
+        TestUtils.waitFor("Wait until resource condition:" + condition + " is fulfilled", Duration.ofSeconds(10).toMillis(), Duration.ofMinutes(16).toMillis(),
```

Comment:
Magic numbers + wrong description, you should rephrase it because `waiting for` is already inside the wait